doc/developers/advanced_installation.rst

```diff
@@ -431,6 +431,12 @@ in the user folder using conda:
     conda activate sklearn-dev
     pip install -v --no-use-pep517 --no-build-isolation -e .
 
+.. note::
+
+   If you installed these packages after creating and activating a new conda
+   environment, you will need to first deactivate and then reactivate the
+   environment for these changes to take effect.
```

Comment:
Instead, I would split the prompt above:

```
.. prompt:: bash $

    conda create -n sklearn-dev -c conda-forge python numpy scipy cython \
        joblib threadpoolctl pytest compilers

It is not always necessary but it safer to open a new prompt before
activating the newly created conda environment.

.. prompt:: bash $
    conda activate sklearn-dev
    pip install -v --no-use-pep517 --no-build-isolation -e .``
```

Reply:
I agree. Should we also add it to other instances ([here](https://github.com/scikit-learn/scikit-learn/blob/04b3a5c5e9a06d5b933ad8ea0aa6cd94c2919a49/doc/developers/advanced_installation.rst?plain=1#L290) and [here](https://github.com/scikit-learn/scikit-learn/blob/04b3a5c5e9a06d5b933ad8ea0aa6cd94c2919a49/doc/developers/advanced_installation.rst?plain=1#L72)) of these commands directly?

Reply:
Yep. I think this would be useful.